pkg/credentials/service_account_credentials.go

```diff
@@ -62,6 +63,58 @@ func NewCredentialBulder(client client.Client, config *v1.ConfigMap) *Credential
 	}
 }
 
+func (c *CredentialBuilder) CreateStorageSpecSecretVolumeAndEnv(namespace string, storageKey string,
+	storageSecretName string, overrideParams map[string]string, container *v1.Container, volumes *[]v1.Volume) error {
+	secret := &v1.Secret{}
+	err := c.client.Get(context.TODO(), types.NamespacedName{Name: storageSecretName,
+		Namespace: namespace}, secret)
+	if err != nil {
+		log.Error(err, "Failed to find secret", "SecretName", storageSecretName)
+		return err
+	}
+	if storageData, ok := secret.Data[storageKey]; ok {
+		var storageDataJson map[string]string
+		err := json.Unmarshal([]byte(storageData), &storageDataJson)
+		if err != nil {
+			return err
+		}
+		// override secret values if parameters are provided
+		for key, element := range overrideParams {
+			storageDataJson[key] = element
+		}
+		if storageDataJson["type"] == "s3" {
+			if endpointURL, ok := storageDataJson["endpoint_url"]; ok {
+				container.Env = append(container.Env, v1.EnvVar{
+					Name:  "AWS_ENDPOINT_URL",
```

Comment:
We should use the existing constants for these env var values (defined in s3_secret.go).

Do we also need to set the `S3_ENDPOINT` and `S3_USE_HTTPS` env vars as is done in `s3_secret.go`?

Reply:
`S3_ENDPOINT` and `S3_USE_HTTPS` is for the minio python client. I think KServe changed the S3 client to boto3 in KFServing 0.5+ due to some AWS IAM is not working with minio python client.

Reply:
So we only need `AWS_ENDPOINT_URL` in this case